docs/commands/push.md

```diff
@@ -1,93 +1,186 @@
 
-If a `package.json` file exists in the folder from which you're executing the `push` command, it will be uploaded as well. Redocly Workflows will use the `@redocly/openapi-cli` version specified in `package.json`.
+1. Active user account in a Redocly Workflows organization.
+1. Active [personal API key](../../workflows/personal-api-keys.md).
 
+## Authentication
 
-<div class="warning">
-If a plugin is referenced in the `.redocly.yaml` file, the `push` command will recursively scan the folder containing the plugin and upload all .js, .json, .mjs and .ts files. Make sure that each plugin has all the required files in its folder, because otherwise they will not be uploaded.
-</div>
+To authenticate to the API registry, you can use the `REDOCLY_AUTHORIZATION` environment variable. It can be set to either your [personal API key](../../workflows/personal-api-keys.md) or to an organization-wide API key (configurable by organization owners in **Redocly Workflows > Org settings > API keys**).
 
+Treat the API keys as secrets and work with them accordingly. Consult the documentation for your CI system to learn more about handling secrets:
 
-### `push` usage
+- [Travis CI documentation](https://docs.travis-ci.com/user/environment-variables/)
+- [CircleCI documentation](https://circleci.com/docs/2.0/env-vars/)
+- [GitHub Actions documentation](https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets)
+- [Jenkins documentation](https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#handling-credentials)
 
+## Usage
 
 ```bash
+openapi push <entrypoint> <destination> [branchName]
+openapi push [--upsert] <entrypoint> <destination> [branchName]
+openapi push 
 openapi push [-u] [--run-id id] <path/to/definition.yaml> <@organization-id/api-name@api-version> [branchName]
 ```
 
+## Options
+
+option           | type      | required?    | default     | description
+-----------------|:---------:|:------------:|:-----------:|------------
+`entrypoint`     | `string`  | yes          | -           | Path to the API definition filename that you want to push to the Redocly API registry. (See [the section below](#entrypoints) for more options)
+`destination`    | `string`  | yes          | -           | Organization id, API name and API version that define the location in the Redocly API registry where to push your API definition.<br />**Format:** `@organization-id/api-name@api-version`. (See [the section below](#destination) for more information)
+`branchName`     | `string`  | no           | `main`      | Branch name that will act as a default one for pushing API definitions to
+`--help`         | `boolean` | no           | -           | Show help
+`--run-id`       | `string`  | no           | -           | CI job ID that the current push will be associated with. See [the section below](#run-id) for more information
+`--upsert`, `-u` | `boolean` | no           | -           | Create a new API and a new API definition version
```

Comment:
Shouldn't this also link to the section with more details? This is an important option because it enables behavior that is not supported by default, and the users will be looking for info how to do it.